platform/ui/src/components/studyBrowser/ImageThumbnail.js

```diff
@@ -48,21 +48,29 @@ export default class ImageThumbnail extends PureComponent {
       cornerstone
         .loadAndCacheImage(this.props.imageId)
         .then(image => {
-          cornerstone.renderToCanvas(this.canvas.current, image);
-          this.setState({
-            loading: false,
-          });
+          if (!this.abortAsyncOperations) {
+            cornerstone.renderToCanvas(this.canvas.current, image);
+            this.setState({
+              loading: false,
+            });
+          }
         })
         .catch(error => {
-          this.setState({
-            loading: false,
-            error: true,
-          });
-          throw new Error(error);
+          if (!this.abortAsyncOperations) {
+            this.setState({
+              loading: false,
+              error: true,
+            });
+            throw new Error(error);
+          }
```

Comment:
Ahh. So these requests have already been initiated; we just need to make sure we're not trying to render to a removed canvas if we've unmounted. We can probably clean this up a bit w/ `async/await`, and just returning early if we have our result.

```js
const { imageId } = this.props;
try {
  const image = await cornerstone.loadAndCacheImage(imageId);

  if (!this.canvas.current) return;
  cornerstone.renderToCanvas(this.canvas.current, image);
  this.setState({ loading: false });
} catch(error) {
  this.setState({ loading: false, error: true });
  throw new Error(error);
}
```

Reply:
Not only this, we must also make sure we do not call `setState` when the component has been unmounted. I found an article about preventing this kind of error and I changed the code to use the _cancelable_ pattern described there:
https://reactjs.org/blog/2015/12/16/ismounted-antipattern.html